assets/helpers/user/userReducer.js

```diff
@@ -48,34 +90,49 @@ function userReducer(
       return Object.assign({}, state, { id: action.id });
 
     case 'SET_DISPLAY_NAME':
-      return Object.assign({}, state, { displayName: action.name });
+      return { ...state, displayName: action.name };
 
     case 'SET_FIRST_NAME':
-      return Object.assign({}, state, { firstName: action.name });
+      return { ...state, firstName: { ...state.firstName, value: action.name } };
 
     case 'SET_LAST_NAME':
-      return Object.assign({}, state, { lastName: action.name });
+      return { ...state, lastName: { ...state.lastName, value: action.name } };
+
 
     case 'SET_FULL_NAME':
-      return Object.assign({}, state, { fullName: action.name });
+      return { ...state, fullName: { ...state.fullName, value: action.name } };
+
 
     case 'SET_TEST_USER':
-      return Object.assign({}, state, { isTestUser: action.testUser });
+      return { ...state, isTestUser: action.testUser };
 
     case 'SET_POST_DEPLOYMENT_TEST_USER':
-      return Object.assign({}, state, { isPostDeploymentTestUser: action.postDeploymentTestUser });
+      return { ...state, isPostDeploymentTestUser: action.postDeploymentTestUser };
 
     case 'SET_EMAIL':
-      return Object.assign({}, state, { email: action.email });
+      return { ...state, email: { ...state.email, value: action.email } };
+
 
     case 'SET_STATEFIELD':
-      return Object.assign({}, state, { stateField: action.stateField });
+      return { ...state, stateField: action.stateField };
 
     case 'SET_GNM_MARKETING':
-      return Object.assign({}, state, { gnmMarketing: action.preference });
+      return { ...state, gnmMarketing: action.preference };
 
     case 'SET_IS_SIGNED_IN':
-      return Object.assign({}, state, { isSignedIn: action.isSignedIn });
+      return { ...state, isSignedIn: action.isSignedIn };
+
+    case 'SET_FIRST_NAME_SHOULD_VALIDATE':
+      return { ...state, firstName: { ...state.firstName, shouldValidate: true } };
+
+    case 'SET_LAST_NAME_SHOULD_VALIDATE':
+      return { ...state, lastName: { ...state.lastName, shouldValidate: true } };
+
+    case 'SET_EMAIL_SHOULD_VALIDATE':
+      return { ...state, email: { ...state.email, shouldValidate: true } };
+
+    case 'SET_FULL_NAME_SHOULD_VALIDATE':
+      return { ...state, fullName: { ...state.fullName, shouldValidate: true } };
```

Comment:
👍 agree with @joelochlann's earlier comment, good call on switching these to object spread.